### PR TITLE
Rewording: Integrations become (store) plugins

### LINKED ISSUE #3594: `Integrations` should be `Store Plugins` and `Plugins` should be `Server Plugins`
Right now we're using quite a few different terms. `Integrations` within the store is basically a store plugin page.

`Store plugins` should show plugins server admin activated  and allow people to access the plugin (click on them), but shouldn't allow adding plugins or disabling them.

`Server plugins` is basically what we now call `plugins` which are available to admins only.

Both pages should allow people to access plugins and open them. Right now we have plugins like whitepaper or hairstylist which once activated nothing happens, because a user isn't aware where they can access them, so allowing easy way to access activated plugin should be had in mind when re-making these pages.

I've discussed this with @Kukks and it seems we'd want to do it for 1.5.0 if we can, otherwise for 1.6.0

### PR DESCRIPTION
**Pull request by dennisreimann**
We're keeping the `store-integrations-list` extension point name for backwards compatibility, otherwise I renamed all "integration" occurences in the views and also the routes.

Note: We still refer to external software like WooCommerce, Drupal, Magento and such as "integrations", I think that distinction makes sense.

Closes #3594.

### CODE REVIEW

**Comment by pavlenex** [comment]
Would it make sense to somehow emphasize these are `Store plugins` vs `Plugins` in the server's side? 
At least maybe in the heading ,but not in the menu? Or add a sentence explaining it through  #3266  later on?

Screenshot 2022-05-10 at 14 08 05

**Comment by pavlenex** [comment]
You're on point @dennisreimann I was actually thinking about the same as soon as I hit that green comment button. Let's keep it unified and see if it causes confusions to admins, though obviously those people aren't just regular users and are a bit tech savvier.

File: BTCPayServer/Views/UIStores/Plugins.cshtml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-@model IntegrationsViewModel
+@model PluginsViewModel
 @{
     Layout = "../Shared/_NavLayout.cshtml";
-    ViewData.SetActivePage(StoreNavPages.Integrations, "Integrations", Context.GetStoreData().Id);
+    ViewData.SetActivePage(StoreNavPages.Plugins, "Plugins", Context.GetStoreData().Id);
```
Comment (Kukks) [inline BTCPayServer/Views/UIStores/Plugins.cshtml:4]: I'm worried this change may break plugins.

Comment (dennisreimann) [reply BTCPayServer/Views/UIStores/Plugins.cshtml:4]: Can we add a deprecation warning and handle it backwards-compatible?

**Comment by dennisreimann** [comment]
@Kukks see [eea52fd](https://github.com/btcpayserver/btcpayserver/pull/3712/commits/eea52fdf68cb44ba3fad1acf6573cd17614ff157).

**Review by Kukks — APPROVED** [review]

**Pull request merged**